src/py-opentimelineio/opentime-bindings/opentime_timeRange.cpp

```diff
@@ -42,6 +42,15 @@ meaning that :meth:`end_time_inclusive` (last portion of a sample in the time ra
                         return TimeRange(*start_time, *duration);
                     }
         }), "start_time"_a=nullptr, "duration"_a=nullptr)
+        .def(py::init<double, double, double>(), "start_time"_a, "duration"_a, "rate"_a)
+        .def("is_invalid_range", &TimeRange::is_invalid_range, R"docstring(
+Returns true if the time range is invalid. The time range is considered invalid if either the start time or
+duration is invalid.
```

Comment:
Could you also document the `duration >= 0` check here?

Reply:
I added comments to the C++ header and Python bindings.